pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.kt

```diff
@@ -126,6 +140,44 @@ class PushPayloadHelper {
     }
 
     class FCMPayload : PushPayloadSerializer {
+        var custom: Map<String, Any>? = null
+        var data: Map<String, Any>? = null
+        var notification: Notification? = null
+
+        override fun toMap(): Map<String, Any> {
+            return mutableMapOf<String, Any>().apply {
+                custom?.let { putAll(it) }
+                data?.let {
+                    if (it.isNotEmpty()) {
+                        put("data", it)
+                    }
+                }
+                notification?.let {
+                    it.toMap().run {
+                        if (this.isNotEmpty()) {
+                            put("notification", this)
+                        }
+                    }
+                }
+            }
+        }
+
+        class Notification : PushPayloadSerializer {
+            var title: String? = null
```

Comment:
It seem it could be extracted since there is a duplication. Notification is defined in FCMPayload and FCMPayloadV2 
?

Reply:
It's intentional - I can't move the `FCMPayload.Notification` class because it would be a breaking change, and I can't use it in `FCMPayloadV2` because `FCMPayload` will be deleted in the future. And in any case after the removal of the deprecated FCMPayload (v1), only FCMPayloadV2.Notification will remain.

Reply:
Ok